src/Microsoft.Data/DataFrame.cs

```diff
@@ -64,17 +64,73 @@ public DataFrame(IList<DataFrameColumn> columns)
         }
 
         /// <summary>
-        /// Returns a new DataFrame using the rows or true values in filterColumn
+        /// Returns a new DataFrame using the boolean values in <paramref name="filter"/>
         /// </summary>
-        /// <param name="filterColumn">A column of rows/bools</param>
-        /// <remarks>filterColumn must be of type long, int or bool</remarks>
-        public DataFrame this[DataFrameColumn filterColumn] => Clone(filterColumn);
+        /// <param name="filter">A column of booleans</param>
+        public DataFrame Filter(PrimitiveDataFrameColumn<bool> filter) => Clone(filter);
```

Comment:
Why have both indexers and `Filter` methods?

Reply:
We don't HAVE to have it. I added it to aid discoverability that we have a "Filter" method. Otherwise it'd be hidden in the indexers, and I'm not certain users will find it easily.

Reply:
Ok, we can always change it later before making this a stable API if we get feedback either way on it.